src/components/chat/chat_channels.jsx

```diff
@@ -51,7 +59,7 @@ const ChatChannels = React.createClass({
 
   getInitialState() {
     return {
-      currentChannel: ChatChannelsStore.getCurrentChannel(),
+      currentChannel: this.getParams().roomSlug,
```

Comment:
@pletcher yes it does, and roomSlug will always be present, otherwise the route wouldn't match.